magpie-run

```diff
@@ -1416,6 +1416,164 @@ else
 fi
 
 #
+# Setup Kafka
+#
+
+if [ "${KAFKA_SETUP}" == "yes" ] && [ "${prior_setup_successful}" == "true" ]
+then
+    cd ${KAFKA_HOME}
+
+    if [ ${KAFKA_MODE} != "setuponly" ]
+    then
+        # Make variables unspecified for launching
+	Magpie_make_all_local_dirs_unspecified
+
+	echo "Starting Kafka"
+        ${MAGPIE_SCRIPTS_HOME}/bin/magpie-kafka.sh start 
+	
+	Magpie_make_all_local_dirs_node_specific
+
+        # My rough estimate for setup time is 30 seconds
+        sleepwait=30
+
```

Comment:
Given kafka depends on Zookeeper, I think that a strict 30 seconds probably isn't good enough.  Could you adjust perhaps similarly to how Storm does it.

I know very little of Kafka, so if there's an underlying thing in Kafka I don't understand, this may be different.  But I assume some type of data is sent to Zookeeper and some quorom has to be reached?